models/biocro/R/write.configs.BIOCRO.R

```diff
@@ -1,92 +1,99 @@
 
-##------------------------------------------------------------------------------------------------#
 ##' convert parameters from PEcAn database default units to biocro defaults
-##' 
+##'
 ##' Performs model specific unit conversions on a list of trait values,
 ##' such as those provided to write.config
 ##' @name convert.samples.BIOCRO
 ##' @title Convert samples for biocro
 ##' @param trait.samples a matrix or dataframe of samples from the trait distribution
+##' @param biocro_version numeric, but currently only checks whether version is less than 1.0
 ##' @return matrix or dataframe with values transformed
 ##' @export
 ##' @author David LeBauer
-convert.samples.BIOCRO <- function(trait.samples) {
-  
+convert.samples.BIOCRO <- function(trait.samples, biocro_version=1.0) {
+
   if (is.list(trait.samples)) {
     trait.samples <- as.data.frame(trait.samples)
   }
-  ## first rename variables
   trait.names <- colnames(trait.samples)
-  trait.names[trait.names == "Vcmax"] <- "vmax"
-  trait.names[trait.names == "Jmax"] <- "jmax"
-  trait.names[trait.names == "leaf_respiration_rate_m2"] <- "Rd"
-  trait.names[trait.names == "cuticular_cond"] <- "b0"
-  trait.names[trait.names == "stomatal_slope.BB"] <- "b1"
-  trait.names[trait.names == "SLA"] <- "Sp"
-  trait.names[trait.names == "growth_respiration_coefficient"] <- "GrowthRespFraction"
-  trait.names[trait.names == "extinction_coefficient_diffuse"] <- "kd"
-  trait.names[trait.names == "chi_leaf"] <- "chi.l"
-  trait.names[trait.names == "quantum_efficiency"] <- "alpha"
-  
-  colnames(trait.samples) <- trait.names
-  ## Partitioning coefficients: especially leaf phenology
-  
-  ## iRhizome iStem ifrRhizome ifrStem
-  
+
   ## transform values with different units cuticular conductance - BETY default is
   ## umol; BioCro uses mol
-  if ("b0" %in% trait.names) {
-    trait.samples$b0 = udunits2::ud.convert(trait.samples$b0, "umol", "mol")
+  if ("cuticular_cond" %in% trait.names) {
+    trait.samples$cuticular_cond = udunits2::ud.convert(trait.samples$cuticular_cond, "umol", "mol")
+  }
+  if ("SLA" %in% trait.names) {
+    trait.samples$SLA = udunits2::ud.convert(trait.samples$SLA, "kg/m2", "g/cm2")
   }
-  if ("Sp" %in% trait.names) {
-    trait.samples$Sp = udunits2::ud.convert(trait.samples$Sp, "kg/m2", "g/cm2")
+
+  ## rename bety variables to match active version of biocro
+  names(trait.names) <- trait.names #looks weird, but naming this vector simplifies indexing below
+
+  name_lookup <- dplyr::tribble(
+    ~bety_name, ~biocro_0.9_name, ~biocro_1.0_name,
+    "Vcmax", "vmax", "vmax1",
+    "Jmax", "jmax", "jmax",
+    "leaf_respiration_rate_m2", "Rd", "Rd",
+    "cuticular_cond", "b0", "b0",
+    "stomatal_slope.BB", "b1", "b1",
+    "SLA", "Sp", "iSp",
+    "growth_respiration_coefficient", "GrowthRespFraction", "growth_respiration_fraction",
+    "extinction_coefficient_diffuse", "kd", "kd",
+    "chi_leaf", "chi.l", "chil",
+    "quantum_efficiency", "alpha", "alpha"
+  )
```

Comment:
@justinmcgrath in addition, this would be a good time to suggest any other parameters that you think would be useful to analyze in sensitivity analysis / variance decomposition or estimate via calibration (e.g. partitioning and allocation parameters?)

Reply:
Ah, well here's a mapping. If you made this more general, you could use it elsewhere.

Potentially we could examine anything that's differentiable.  For now, I've changes the growth model and would be interested in these parameters.

resistance_leaf_to_stem
resistance_stem_to_grain
resistance_stem_to_root
resistance_stem_to_rhizome
rate_constant_leaf_senescence
rate_constant_stem_senescence
rate_constant_root_senescence
rate_constant_rhizome_senescence
KmLeaf_senescence
KmStem_senescence
KmRoot_senescence
KmRhizome_senescence

Reply:
@justinmcgrath for these could you please fill out this table so that I can add these to the table: https://docs.google.com/spreadsheets/d/1GVPcEtcj2oTMT6N_4CUK0nX3VBOjkXOLTo5Rzz4MH3Y/edit?usp=sharing

Note: 
* min and max (optional) should be the range of possible values (generally lower bound is 0, this is for crude qa/qc)
* the prior distribution, param a, param b follow R conventions in https://cran.r-project.org/doc/manuals/R-intro.html#R-as-a-set-of-statistical-tables